### PR TITLE
mapl: add 2.48.0

### PR DESCRIPTION
**Pull request by mathomp4**
This update adds MAPL 2.48.0. Also adds `udunits` as a dependency and adds a good check method.

### CODE REVIEW

File: var/spack/repos/builtin/packages/mapl/package.py [hunk 1]
```diff
@@ -282,16 +293,19 @@ class Mapl(CMakePackage):
     depends_on("pflogger@1.9.1: +mpi", when="@2.23:2.39+pflogger")
     depends_on("pflogger@1.9.5: +mpi", when="@2.40:2.43+pflogger")
     depends_on("pflogger@1.11.0: +mpi", when="@2.44+pflogger")
-    depends_on("pflogger@1.14.0: +mpi", when="@2.45:+pflogger")
+    depends_on("pflogger@1.14.0: +mpi", when="@2.45:2.47+pflogger")
```
Comment (climbfuji) [inline var/spack/repos/builtin/packages/mapl/package.py]: All other dependencies are order from newest to oldest, except pflogger

File: var/spack/repos/builtin/packages/mapl/package.py [hunk 2]
```diff
@@ -257,14 +264,18 @@ class Mapl(CMakePackage):
...
+    depends_on("gftl@1.14.0:", when="@2.48:")
+    depends_on("gftl@1.13.0:", when="@2.45:2.47")
     depends_on("gftl@1.11.0:", when="@2.44")
     depends_on("gftl@1.10.0:", when="@2.40:2.43")
     depends_on("gftl@1.5.5:1.9", when="@:2.39")
 
     # There was an interface change in gftl-shared, so we need to control versions
     # MAPL 2.39 and older can use up to 1.6.0 but MAPL 2.40+ needs 1.6.1 or higher
-    depends_on("gftl-shared@1.8.0:", when="@2.45:")
+    depends_on("gftl-shared@1.8.0:", when="@2.48:")
```
Comment (climbfuji) [inline var/spack/repos/builtin/packages/mapl/package.py]: Is this a typo? If 2.45:2.47 depend on 1.8.0 and 2.48 as well, these lines can be combined

Comment (mathomp4) [reply var/spack/repos/builtin/packages/mapl/package.py]: Nuts. Yeah bad copy paste!

File: var/spack/repos/builtin/packages/mapl/package.py (hunk 1 above)
Comment (mathomp4) [reply var/spack/repos/builtin/packages/mapl/package.py]: Will fix up.

**Review by climbfuji — APPROVED** [review]
Looks good to me. I verified the checksum, but didn't run the new tests.

**Comment by mathomp4** [comment]
> Looks good to me. I verified the checksum, but didn't run the new tests.

The delay was me trying to get them to run successfully. But:
```
100% tests passed, 0 tests failed out of 33
```
Woo!

**Review by bernhardkaindl — APPROVED** [review]
Built on Ubuntu 22.04

**Pull request merged**